### PR TITLE
made loading...

### PR DESCRIPTION
**Pull request by hadipournigjeh**
- Update Home.js
- Update input.js

### CODE REVIEW

File: client/src/pages/Home.js [hunk 1]
```diff
@@ -72,12 +72,41 @@ const Home = () => {
...
+				setShowOutput(true);
+			} else {
+				console.log("Error: " + response.status);
+			}
+		} catch (error) {
+			console.log(error);
+		} finally {
+			setIsLoading(false);
+			setInputValue("");
+		}
```
Comment (LexiisYing) [inline client/src/pages/Home.js:109]: The fetch is already done in `Input.js` component, it is not necessary to fetch again in here @hadipournigjeh

File: client/src/pages/Home.js [hunk 2]
```diff
@@ -72,12 +72,41 @@ const Home = () => {
...
-	const handleSubmit = (data) => {
-		setOutputValue(data);
-		setShowOutput(true);
+	const handleSubmit = () => {
+		if (inputValue.trim() !== "") {
+			setIsLoading(true); // Set loading state to true
+			fetchData();
+		}
+	};
+
```
Comment (LexiisYing) [inline client/src/pages/Home.js:85]: I think we can keep the original handleSubmit logic, and we don't need to set loading in here

File: client/src/pages/Home.js [hunk 3]
```diff
@@ -31,13 +31,13 @@ const Home = () => {
 		})
 			.then((response) => {
 				if (response.ok) {
-					return response.json(); // Parse the response data as JSON
+					return response.json();
```
Comment (LexiisYing) [inline client/src/pages/Home.js:34]: Not necessary to remove this comment

File: client/src/pages/Home.js [hunk 4]
```diff
@@ -31,13 +31,13 @@ const Home = () => {
...
 				} else {
 					throw new Error("Failed to save user data");
 				}
 			})
 			.then((data) => {
-				console.log(data.message); // Log the response message from the backend
```
Comment (LexiisYing) [inline client/src/pages/Home.js:40]: Not necessary to remove this comment

**Review by LexiisYing — CHANGES_REQUESTED** [review]

File: client/src/pages/Home.js (hunk 1 above)
Comment (hadipournigjeh) [reply client/src/pages/Home.js:109]: Hi Lexi , unfortunately I made it this one and already it works , I don't have new ideas to change again :-(

**Pull request closed without merging**